docs/eta-pi-p/manual.ipynb

```diff
@@ -255,24 +276,235 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Visualization"
+    "### Kinematic variables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@unevaluated\n",
+    "class SquaredInvariantMass(sp.Expr):\n",
+    "    momentum: sp.Basic\n",
+    "    _latex_repr_ = \"m_{{{momentum}}}^2\"\n",
+    "\n",
+    "    def evaluate(self) -> sp.Expr:\n",
+    "        p = self.momentum\n",
+    "        p_xyz = ThreeMomentum(p)\n",
+    "        return Energy(p) ** 2 - EuclideanNorm(p_xyz) ** 2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "source_hidden": true
+    }
+   },
+   "outputs": [],
+   "source": [
+    "p1 = FourMomentumSymbol(\"p1\", shape=[])\n",
+    "p2 = FourMomentumSymbol(\"p2\", shape=[])\n",
+    "p3 = FourMomentumSymbol(\"p3\", shape=[])\n",
+    "p12 = ArraySum(p1, p2)\n",
+    "p23 = ArraySum(p2, p3)\n",
+    "p31 = ArraySum(p3, p1)\n",
+    "\n",
+    "theta12 = Theta(p12)\n",
+    "theta23 = Theta(p23)\n",
+    "theta31 = Theta(p31)\n",
+    "\n",
+    "phi12 = Phi(p12)\n",
+    "phi23 = Phi(p23)\n",
+    "phi31 = Phi(p31)\n",
```

Comment:
These are in themselves already readable (and there is no performance hit to recompute them if you use them again). So I would not define them as separate variables, but just call `Phi(p12)` directly.